app/uk/gov/hmrc/transitmovementsvalidator/services/JsonValidationService.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsvalidator.services
+
+import akka.stream.Materializer
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.StreamConverters
+import akka.util.ByteString
+import cats.data.NonEmptyList
+import com.google.inject.ImplementedBy
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.networknt.schema.JsonSchema
+import com.networknt.schema.JsonSchemaFactory
+import com.networknt.schema.SpecVersion
+import com.networknt.schema.ValidationMessage
+import uk.gov.hmrc.transitmovementsvalidator.models.MessageType
+import uk.gov.hmrc.transitmovementsvalidator.models.errors.JsonSchemaValidationError
+import uk.gov.hmrc.transitmovementsvalidator.models.errors.ValidationError
+
+import javax.inject.Inject
+import scala.collection.JavaConverters.asScalaSetConverter
+
+@ImplementedBy(classOf[JsonValidationServiceImpl])
+trait JsonValidationService {
+
+  def validate(messageType: String, source: Source[ByteString, _])(implicit
+    materializer: Materializer,
+    ec: ExecutionContext
+  ): Future[Either[NonEmptyList[ValidationError], Unit]]
+}
+
+class JsonValidationServiceImpl @Inject() extends JsonValidationService {
+
+  private val mapper  = new ObjectMapper
+  private val factory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7)
```

Comment:
Not important for now, but at some point, might want to make the version an external property.

Reply:
You'd think it could auto select based on the fact the version is specified in the Json schema... alas...